### DISCUSSION TITLE
Read only mode for one projects for a customer

### DISCUSSION THREAD

**Discussion by TheZoker** [kimai/kimai#4378] [Questions & Answers]
Hi there,

I would like to give one of my customers access to my time tracking. 
He should:
- be able to login (so he needs a user account I guess)
- not be able to create a time entry (similar to a read only mode)
- only be able to see "his" projects (all projects assigned to this customer)
- see my time tracking entries

Currently I'm able to create a team for this customer and assign his user account to this team and the client to this team as well. But the account is able to create a new time entry, where the user can see all the projects and other customers trough the client selection dropdown:

![image](https://github.com/kimai/kimai/assets/1368405/649bf2f0-694d-4966-b668-03be17271eca)

He is also not able to see my time tracking (this can maybe be resolved with a custom role?).

What is the best way to achieve this? Is it possible with already existing functions?

**Comment by TheZoker** [comment]
I just found this plugin:
https://www.kimai.org/store/derpilzcc-readonly-access-bundle.html

Will try it out and give feedback :)

**Reply by TheZoker** [reply]
Sadly this plugin does not seem to be compatible with version 2

**Reply by TheZoker** [reply]
@kevinpapst Would it be hard to make this plugin compatible with the new kimai version?

**Comment by enbyted** [comment]
I recently set it up. It was relatively simple, though needed quite a bit of fiddling around the permissions. 
In the end the following points wre important:
 - Create role for normal users and copy permissions from ROLE_USER to it
 - Remove all permissions from ROLE_USER (well, I left out the viewing and editing of own profile)
 - Make sure that all your customers have at least one team -> ones that do not will be globally visible
 - Create a new role (I called it ROLE_SPECTATOR) and set whatever permissions you need, I provided activity, project and client view_ and time_ for team member
 - Create a user with ROLE_SPECTATOR, assign it to selected team(s)

And viola, it works.
You can even disable the view own timesheet permission, if you set the default landing page to be something that the user has access to.
I was able to configure everything via the UI.
